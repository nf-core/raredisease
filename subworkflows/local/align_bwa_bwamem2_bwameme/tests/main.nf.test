nextflow_workflow {

    name "Test Workflow ALIGN_BWA_BWAMEM2_BWAMEME"
    script "subworkflows/local/align_bwa_bwamem2_bwameme/main.nf"
    workflow "ALIGN_BWA_BWAMEM2_BWAMEME"
    tag "subworkflows"
    tag "align_bwa_bwamem2_bwameme"
    config "./nextflow.config"

    test("align bwamem2") {

        setup {
            run("BWAMEM2_INDEX") {
                script "modules/nf-core/bwamem2/index/main.nf"
                process {
                    """
                    input[0] = Channel.of([
                        [id:'sarscov2'],
                        file('https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                    ])
                    """
                }
            }
        }

        when {
            params {
                aligner = "bwamem2"
                sarscov_testdata_base_path= 'https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/'
                save_mapped_as_cram = true
                restrict_to_contigs = "MT192765.1"
            }
            workflow {
                """
                input[0] = Channel.of([
                    [ id:'test', sample:'test', single_end:false, num_lanes:1, read_group: "'@RG\\\\tID:test\\\\tPL:illumina\\\\tSM:test'"  ], // meta map
                    [
                        file(params.sarscov_testdata_base_path+ 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                        file(params.sarscov_testdata_base_path+ 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkIfExists: true)
                    ]
                ])
                input[1] = [[:],[]]
                input[2] = BWAMEM2_INDEX.out.index
                input[3] = [[:],[]]
                input[4] = Channel.of([[id:'sarscov2'], [file(params.sarscov_testdata_base_path+ 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)]])
                input[5] = Channel.of([[id:'sarscov2'], [file(params.sarscov_testdata_base_path+ 'genomics/sarscov2/genome/genome.fasta.fai', checkIfExists: true)]])
                input[6] = 3072
                input[7] = "illumina"
                input[8] = 4
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                        workflow.out.marked_bam.collect { meta, bamfile -> [ meta, bam(bamfile).getHeaderMD5() ] },
                        workflow.out.marked_bam.collect { meta, bamfile -> [ meta, bam(bamfile).getReadsMD5() ] },
                        workflow.out.versions
                    ).match()
                }
            )
        }
    }

    test("align bwameme") {

        setup {
            run("BWAMEME_INDEX") {
                script "modules/nf-core/bwameme/index/main.nf"
                process {
                    """
                    input[0] = Channel.of([
                        [id:'sarscov2'],
                        file('https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                    ])
                    """
                }
            }
        }

        when {
            params {
                aligner = "bwameme"
                sarscov_testdata_base_path= 'https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/'
                save_mapped_as_cram = false
                restrict_to_contigs = "MT192765.1"
            }
            workflow {
                """
                input[0] = Channel.of([
                    [ id:'test', sample:'test', single_end:false, num_lanes:1, read_group:"\'@RG\\\\tID:test\\\\tPL:illumina\\\\tSM:test\'"  ], // meta map
                    [
                        file(params.sarscov_testdata_base_path+ 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                        file(params.sarscov_testdata_base_path+ 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkIfExists: true)
                    ]
                ])
                input[1] = [[:],[]]
                input[2] = [[:],[]]
                input[3] = BWAMEME_INDEX.out.index
                input[4] = Channel.of([[id:'sarscov2'], [file(params.sarscov_testdata_base_path+ 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)]])
                input[5] = Channel.of([[id:'sarscov2'], [file(params.sarscov_testdata_base_path+ 'genomics/sarscov2/genome/genome.fasta.fai', checkIfExists: true)]])
                input[6] = 3072
                input[7] = "illumina"
                input[8] = 4
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                        workflow.out.marked_bam.collect { meta, bamfile -> [ meta, bam(bamfile).getHeaderMD5() ] },
                        workflow.out.marked_bam.collect { meta, bamfile -> [ meta, bam(bamfile).getReadsMD5() ] },
                        workflow.out.versions
                    ).match()
                }
            )
        }
    }

}
