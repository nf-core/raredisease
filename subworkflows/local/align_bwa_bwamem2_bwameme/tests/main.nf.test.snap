{
    "align bwamem2": {
        "content": [
            [
                [
                    {
                        "groupSize": 1,
                        "groupTarget": {
                            "id": "test",
                            "sample": "test",
                            "single_end": false,
                            "num_lanes": 1,
                            "read_group": "'@RG\\tID:test\\tPL:illumina\\tSM:test'"
                        }
                    },
                    "17f8c2ac941e887d9470f69453c08ea5"
                ]
            ],
            [
                [
                    {
                        "groupSize": 1,
                        "groupTarget": {
                            "id": "test",
                            "sample": "test",
                            "single_end": false,
                            "num_lanes": 1,
                            "read_group": "'@RG\\tID:test\\tPL:illumina\\tSM:test'"
                        }
                    },
                    "af8628d9df18b2d3d4f6fd47ef2bb872"
                ]
            ],
            [
                "versions.yml:md5,6046f0e9edf48b8d3bd335da75134ee4",
                "versions.yml:md5,952f0145e6a0a3d1dbd70aac2edf4ced",
                "versions.yml:md5,9d7492d23e5995a4955de5f751df05af",
                "versions.yml:md5,e9e514dbea5a822d1978acac57bee61e",
                "versions.yml:md5,f56a243ff21d59d873e0d0a3d3ee441d"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-05-23T16:03:29.762040277"
    },
    "align bwameme": {
        "content": [
            [
                [
                    {
                        "groupSize": 1,
                        "groupTarget": {
                            "id": "test",
                            "sample": "test",
                            "single_end": false,
                            "num_lanes": 1,
                            "read_group": "'@RG\\tID:test\\tPL:illumina\\tSM:test'"
                        }
                    },
                    "144054510dda19100c4274a7db41bd8"
                ]
            ],
            [
                [
                    {
                        "groupSize": 1,
                        "groupTarget": {
                            "id": "test",
                            "sample": "test",
                            "single_end": false,
                            "num_lanes": 1,
                            "read_group": "'@RG\\tID:test\\tPL:illumina\\tSM:test'"
                        }
                    },
                    "af8628d9df18b2d3d4f6fd47ef2bb872"
                ]
            ],
            [
                "versions.yml:md5,390a09ae83280da84de93856149cc3a1",
                "versions.yml:md5,952f0145e6a0a3d1dbd70aac2edf4ced",
                "versions.yml:md5,9d7492d23e5995a4955de5f751df05af",
                "versions.yml:md5,e9e514dbea5a822d1978acac57bee61e",
                "versions.yml:md5,f56a243ff21d59d873e0d0a3d3ee441d"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-05-23T16:04:03.379890309"
    }
}